threads/server-tcp.c

```diff
@@ -0,0 +1,154 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <unistd.h>
+#include <inttypes.h>
+#include <string.h>
+#include <pthread.h>
+
+/****************************************
+        Author: Tim Wood
+        Co-Sub-Authors: Eric Armbrust, Neel Shah, Phil Lopreiato
+        with a little help from
+        http://beej.us/guide/bgnet/
```

Comment:
If you have any useful references for threads you can post them here.  "man pthread" is also fine.